packages/core/botpress/src/cli/ghost-sync.js

```diff
@@ -31,7 +31,16 @@ const writeRevisions = async (revisionsFile, revisions) => {
 
 const writeFile = folderPath => async ({ file, content, deleted }) => {
   const filePath = path.join(folderPath, file)
-  return deleted ? fs.unlinkAsync(filePath) : fs.writeFileAsync(filePath, content)
+
+  try {
+    await fs.accessAsync(filePath)
+    return deleted ? fs.unlinkAsync(filePath) : fs.writeFileAsync(filePath, content)
+  } catch (e) {
+    if (deleted) {
+      return Promise.resolve(true)
+    }
+    return fs.writeFileAsync(filePath, content)
+  }
 }
```

Comment:
I feel the same can be written in a simpler way. For example, this code will fail if you don't have the write permission as you'll try to write this file twice, second time in the catch clause.

Is this simpler?

```
if (deleted) {
  try {
    fs.unlinkAsync(filePath)
  } catch (e) {
    return Promise.resolve(true)
  }
} else {
  return fs.writeFileAsync(filePath, content)
}
```

Or better only return `true` if the file actually does not exist (check the `accessAsync` exception code), otherwise let it fail normally.

Reply:
OK, makes sense, will change that.

Reply:
@emirotin , would it look better this way?